joint_state_controller/src/joint_state_controller.cpp

```diff
@@ -38,37 +38,55 @@ class State;
 
 namespace joint_state_controller
 {
-static const auto kPositionIndex = 0ul;
-static const auto kVelocityIndex = 1ul;
-static const auto kEffortIndex = 2ul;
+const auto kUninitializedValue = std::numeric_limits<double>::quiet_NaN();
+const auto kPositionName = "position";
+const auto kVelocityName = "velocity";
+const auto kEffortName = "effort";
```

Comment:
I propose to use names from:  hardware_interface/include/hardware_interface/types/hardware_interface_type_values.hpp

```suggestion
```

Reply:
at the time of first doing this, these values were not available. Replaced with this constants now